### PR TITLE
JBR-7072 Wayland: clicks on items of floating context menus are ignored

### PR DESCRIPTION
**Pull request by mkartashev**
[JBR-7072](https://youtrack.jetbrains.com/issue/JBR-7072) Wayland: clicks on items of floating context menus are ignored

The origin of the problem lies in an unexpected sequence of FocusEvent's generated on the popup invocation. In order to make that sequence more standard, I used the corresponding method from LWComponentPeer as a template. It doesn't cover all Wayland cases yet and falls through to `WLKeyboardFocusManagerPeer.SNFH_FAILURE` case in some instances, but that's a task for another day.

Also verified that this fix did not regress on the test from [JBR-6145](https://youtrack.jetbrains.com/issue/JBR-6145) [Wayland toolkit] Popup windows aren't focusable.

### CODE REVIEW

**Comment by JB-Dmitry** [comment]
Worth creating an automated test case? Or at least a simple reproducer for manual checking, not requiring to launch the IDE?

**Comment by mkartashev** [comment]
> Worth creating an automated test case? Or at least a simple reproducer for manual checking, not requiring to launch the IDE?

Yes, I attached a manual test to the [issue here](https://youtrack.jetbrains.com/issue/JBR-7072/Wayland-clicks-on-items-of-floating-context-menus-are-ignored#focus=Comments-27-9922597.0-0).

**Comment by avu** [comment]
> > Worth creating an automated test case? Or at least a simple reproducer for manual checking, not requiring to launch the IDE?
> 
> Yes, I attached a manual test to the [issue here](https://youtrack.jetbrains.com/issue/JBR-7072/Wayland-clicks-on-items-of-floating-context-menus-are-ignored#focus=Comments-27-9922597.0-0).

I think that a manual jtreg test would be useful here. We have too many reproducers inside the issue without at least a semi-automated way to verify them.

**Comment by mkartashev** [comment]
I added a manual test. It can be run on Wayland like so
```
$ jtreg -testjdk:`pwd`/build/linux-x86_64-server-release/images/jdk -javaoptions:'-Dawt.toolkit.name=WLToolkit' test/jdk/jb/java/awt/Focus/ContextMenuFocusTest.java
```

File: test/jdk/jb/java/awt/Focus/ContextMenuFocusTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 JetBrains s.r.o.
```
Comment (avu) [inline test/jdk/jb/java/awt/Focus/ContextMenuFocusTest.java:2]: Usually we use 
* Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
as well

**Review by avu — APPROVED** [review]
LGTM

**Review by JB-Dmitry — APPROVED** [review]

**Pull request merged**